sklearn/utils/validation.py

```diff
@@ -1655,6 +1655,24 @@ def check_is_fitted(estimator, attributes=None, *, msg=None, all_or_any=all):
         raise NotFittedError(msg % {"name": type(estimator).__name__})
 
 
+def _estimator_has(attr, delegate_attrs=("estimator_", "estimator")):
+    """Check if we can delegate a method to the underlying estimator.
+
+    First, we check the fitted estimator if available, otherwise we
+    check the unfitted estimator.
```

Comment:
```suggestion
    We check the `delegate_attrs` in the order they are passed.
    By default, we first check the fitted estimator if available,
    otherwise we check the unfitted estimator.
```
I think here we should mention that we check any delegates that are passed.